### ISSUE TITLE
DO schema validation is failing in golang

### ISSUE THREAD

**Issue by papineni87** [F5Networks/f5-declarative-onboarding#132]
In general for AS3/DO schema validation we use visual studio to validate our AS3/DO json against schema.we have to implement the same schema validation in our terraform provider and we use  “gojsonschema” a third party golang package to do the validation.
```
root@terraforn-ubuntu3:~/go/src/github.com/terraform-providers/terraform-provider-bigip# cat test1.go 
package main
 
import (
      "fmt"
      "github.com/xeipuuv/gojsonschema"
)
func main() {
    schemaLoader := gojsonschema.NewReferenceLoader("file:///root/go/src/github.com/terraform-providers/terraform-provider-bigip/do-base-schema1.json")
    documentLoader := gojsonschema.NewReferenceLoader("file:///root/go/src/github.com/terraform-providers/terraform-provider-bigip/example1.json")
    result, err := gojsonschema.Validate(schemaLoader, documentLoader)
    if err != nil {
        panic(err.Error())
    }
}
 ```
When we tried above program to validate DO json with DO base schema, we got below error.
 ```
root@terraforn-ubuntu3:~/go/src/github.com/terraform-providers/terraform-provider-bigip# go run test1.go 
2020/03/27 03:24:29 pattern is : pattern
2020/03/27 03:24:29 value is  : ^.{0,254}$
2020/03/27 03:24:29 vlaue of m is :map[description:Password for username account.  This is generally not required to configure 'localhost' and is not required when you populate tokens pattern:^.{0,254}$ type:string]
2020/03/27 03:24:29 pattern is : pattern
2020/03/27 03:24:29 value is  : ^[^\x00-\x1f\x7f-\xff]{0,8192}$
2020/03/27 03:24:29 vlaue of m is :map[pattern:^[^\x00-\x1f\x7f-\xff]{0,8192}$ type:string]
panic: pattern must be a valid regex
```

It looks like current DO schema works fine when Validating in Visual studio, but when we use the same schema for validation in golang, it throws above error. It took time to add debug in third party package to find out the culprit in the schema. When we remove token regular expression in base schema, above program validates DO json against DO base schema.

Below regex used in token is failing, and I am still surprising how it is passing in visual studio. We tested regex in python/golang with valid token and it is failing.

```
"tokens": {
--
  | "description": "One or more HTTP headers (each a property, like 'X-F5-Auth-Token': 'MF6APSRUYKTMSDBEOOEWLCNSO2') you want to send with queries to the device management service as authentication/authorization tokens",
  | "type": "object",
  | "patternProperties": {
  | "^[^\\x00-\\x20:\\x7f-\\xff]{1,254}$": {
  | "type": "string",
  | "pattern": "^[^\\x00-\\x1f\\x7f-\\xff]{0,8192}$"
  | }
  | },
```
If we remove tokens from base schema, schema validation works fine.

**Comment by dstokesf5** [comment]
This is caused by a limitation in Go's regular expression library that does not like repeating a pattern more than 1000 times. I have created AUTOTOOL-1440 to rework this to be more compatible with Go's regular expressions.

**Comment by dstokesf5** [comment]
This issue has been resolved as part of the 1.12.0 release.
